src/adam_core/propagator/adam_assist.py

```diff
@@ -517,10 +523,127 @@ def _detect_impacts(
                 variant_id=[],
             )
         return results, earth_impacts
-
+    
+    def _add_light_time(
+        self, 
+        orbits,
+        observers,
+        lt_tol: float = 1e-10,
+    ):  
+        orbits_aberrated = orbits.empty()
+        lts = np.zeros(len(orbits))
+        for i, (orbit, observer) in enumerate(zip(orbits, observers)):
+            lt0 = 0
+            dlt = 1
+            while dlt > lt_tol:
+                observer_position = observer.coordinates.values[0,:3]
+                orbit_i = orbit
+                t0 = orbit_i.coordinates.time.mjd()[0].as_py()
+
+                rho = np.linalg.norm(orbit_i.coordinates.values[0,:3] - observer_position)
+
+                lt = rho / C
+
+                dlt = np.abs(lt - lt0)
+
+                t1 = t0 - lt
+                t1 = Timestamp.from_mjd([t1], scale="tdb")
+                orbit_propagated = self._propagate_orbits(orbit, t1)
+
+                orbit_i = orbit_propagated
+                t0 = t1
+                lt0 = lt
+                dlt = dlt
+            orbits_aberrated = qv.concatenate([orbits_aberrated, orbit])
+            lts[i] = lt
+
+        return orbits_aberrated, lts
+    
     def _generate_ephemeris(
-        self, orbits: OrbitType, observers: ObserverType
+        self, orbits: OrbitType, observers: ObserverType, lt_tol: float = 1e-10
     ) -> EphemerisType:
-        raise NotImplementedError(
-            "ASSISTPropagator does not yet support ephemeris generation."
-        )
+        
+        if isinstance(orbits, Orbits):
+            ephemeris_total = Ephemeris.empty()
+        elif isinstance(orbits, VariantOrbits):
+            ephemeris_total = VariantEphemeris.empty()
+        
+        for orbit in orbits:
+            propagated_orbits = self._propagate_orbits(orbit, observers.coordinates.time)
```

Comment:
```suggestion
            propagated_orbits = self.propagate_orbits(orbit, observers.coordinates.time)
```
We should be calling the public API here, I think.